src/common/helpers/mediaHelpers.ts

```diff
@@ -291,6 +293,78 @@ export class CloudinaryHelper {
     public static isCloudinaryImage(url: string) {
         return CloudinaryHelper.cloudinaryUrlRegex.test(url);
     }
+
+    async uploadFileInChunks(
+        file: File,
+        uploadData: UploadUrlResponseV1,
+        onSuccess: Function,
+        onFail: Function,
+        onProgress?: (file: File, percent: number) => void
+        ){
+            const chunkSize = 10 * 1024 * 1024; // 10MB
+            const uniqueUploadId = Math.random().toString(36).substring(2, 15);
+    
+                let bytesChunked = 0;
+                const uploadPromises = [];
+                const uploadBytesTrace: number[] = [];
+                const controller = new AbortController();
+                let iteration = 0;
+    
+                while (bytesChunked < file.size && !controller.signal.aborted) {
+                    const chunkIndex = iteration++;
+    
+                    const chunkEnd = Math.min(bytesChunked + chunkSize, file.size);
+                    const chunk = file.slice(bytesChunked, chunkEnd);
+                    const formData = new FormData();
+                    formData.set("file", chunk, file.name);
+                    
+                    Object.entries(uploadData.params).forEach(([key, value]) => {
+                        // @ts-ignore
+                        formData.set(key, value);
+                    });
+                    
+                    const contentRange = `bytes ${bytesChunked}-${chunkEnd-1}/${file.size}`;
+                    const result = sendXhrRequest({
+                        url: uploadData.put_url,
+                        method: "POST",
+                        headers: {
+                            "X-Unique-Upload-Id": uniqueUploadId,
+                            "Content-Range": contentRange,
+                        },
+                        data: formData,
+                        progressCallback: (bytesLoaded: number) => {
+                            if(!onProgress) return;
+                            
+                            uploadBytesTrace[chunkIndex] = bytesLoaded;
```

Comment:
try making this calculation with kind of throttling (no need to update the browser ui if there is no change in percentage)

Reply:
make sure its not very complex (if it is just drop it)